apps/mymedicare_cb/authorization.py

```diff
@@ -66,3 +81,121 @@ def exchange(self, code, request):
+            "client_id": self.client_id,
+            "client_secret": self.client_secret,
+        }
+        headers = {"Content-Type": "application/json",
+                   "X-SLS-starttime": str(datetime.datetime.utcnow())}
+
+        if request is not None:
+            headers.update({"X-Request-ID": str(getattr(request, '_logging_uuid', None)
+                            if hasattr(request, '_logging_uuid') else '')})
+
+        # Get auth flow trace session values dict.
+        auth_flow_dict = get_session_auth_flow_trace(request)
+
+        response = requests.post(
+            self.token_endpoint,
+            auth=self.basic_auth(),
+            json=data_dict,
+            headers=headers,
+            verify=self.verify_ssl,
+            hooks={
+                'response': [
+                    response_hook_wrapper(sender=SLSxTokenResponse,
+                                          auth_flow_dict=auth_flow_dict)]})
+
+        response.raise_for_status()
+
+        token_response = response.json()
+
+        auth_token = token_response.get('auth_token', None)
+        if auth_token is None:
+            log_authenticate_start(auth_flow_dict, "FAIL",
+                                   "Exchange auth_token is missing in response error")
+            raise BBMyMedicareSLSxTokenException(settings.MEDICARE_ERROR_MSG)
+
+        user_id = token_response.get('user_id', None)
+        if user_id is None:
+            log_authenticate_start(auth_flow_dict, "FAIL",
+                                   "Exchange user_id is missing in response error")
+            raise BBMyMedicareSLSxTokenException(settings.MEDICARE_ERROR_MSG)
+
+        return auth_token, user_id
+
+    def auth_header(self):
+        return {"Authorization": "Bearer %s" % (self.token['access_token'])}
+
+    def get_user_info(self, access_token, user_id, request):
+        headers = {"Authorization": "Bearer %s" % (access_token)}
+        # keep using deprecated conv - no conflict issue
+        headers.update({"X-SLS-starttime": str(datetime.datetime.utcnow())})
+
+        if request is not None:
+            headers.update({"X-Request-ID": str(getattr(request, '_logging_uuid', None)
+                            if hasattr(request, '_logging_uuid') else '')})
+
+        # Get auth flow session values.
+        auth_flow_dict = get_session_auth_flow_trace(request)
+        try:
+            response = requests.get(self.userinfo_endpoint + "/" + user_id,
+                                    headers=headers,
+                                    verify=self.verify_ssl,
+                                    hooks={
+                                        'response': [
+                                            response_hook_wrapper(sender=SLSxUserInfoResponse,
+                                                                  auth_flow_dict=auth_flow_dict)]})
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
```

Comment:
Is there some edge case this catches that the `raise_for_status` function call above does not?